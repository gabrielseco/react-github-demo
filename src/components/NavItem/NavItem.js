// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './NavItem.scss';

const NavItem = ({ URL, children }) => (
  <li className={styles.navItem}>
    <Link className={styles.navLink} to={URL}>
      {children}
    </Link>
  </li>
);

export default NavItem;
