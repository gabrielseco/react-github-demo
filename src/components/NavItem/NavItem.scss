$nav-item-link-color: #212240;

.navItem {
  display: inline-block;
  list-style-type: none;
  margin: 0 20px;
}

.navLink {
  color: $nav-item-link-color;
}
