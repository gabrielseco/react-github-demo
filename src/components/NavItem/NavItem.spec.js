import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import NavItem from './NavItem';

describe('NavItem suite', () => {
  it('should render the NavItem component', () => {
    const url = '/repos';
    const name = 'Repos';
    const navItem = <NavItem URL={url}>{name}</NavItem>;
    const navItemToSnapshot = <Router>{navItem}</Router>;
    const component = shallow(navItem);
    const tree = renderer.create(navItemToSnapshot).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();

    expect(component.find(Link).prop('to')).toBe(url);
    expect(component.find(Link).prop('children')).toBe(name);
  });
});
