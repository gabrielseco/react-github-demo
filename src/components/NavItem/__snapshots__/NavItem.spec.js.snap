// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem suite should render the NavItem component 1`] = `
<li
  className="navItem"
>
  <a
    className="navLink"
    href="/repos"
    onClick={[Function]}
  >
    Repos
  </a>
</li>
`;
